provd/persist/util.py

```diff
@@ -552,3 +551,24 @@ def __init__(self, collection):
 
     def __getattr__(self, name):
         return getattr(self._collection, name)
+
+
+def _new_key_fun_from_key(key):
+    """
+    Return a function usable for the key parameter of the sorted function
+    from a [sort] key
+    """
+    split_key = key.split('.')
+
+    def func(document):
+        cur_elem = document
+        try:
+            for cur_key in split_key:
+                cur_elem = cur_elem[cur_key]
+        except (KeyError, TypeError) as e:
+            return False, str(cur_elem)
+        # if the value is not a str/int/float, then return (True, str(cur_elem))
+        # as False < True, the value that is not a str/int/float is put at the end
+        return type(cur_elem) not in (str, int, float), str(cur_elem)
```

Comment:
I'm not sure how this is supposed to work? Why return a tuple for the sorting function?

Reply:
Tuples can be a good way to sort things when you have multiple fields or conditions to sort on. Eg. `(not is_featured, name)` would sort put featured items first and then go alphabetically (since False is sorted before True). 

However, in this does seem a little complicated and I don't see why we want to place certain types before other types.
I only suggested `return '' if cur_element is None else cur_element` to avoid the `or ''` incorrectly handling Falsey values. A side from that your previous idea seemed to work well for all types other than lists/dicts. Which I don't think come through here.

Reply:
After discussion with @sopelj,  I keep the "old" behaviour. The only main new thing is that the value is always returned as a string, to avoid errors like builtins.TypeError: '<' not supported between instances of 'int' and 'str'